index.php

```diff
@@ -40,79 +39,76 @@
-    // User must be authenticated - can see the logged in user's records only
+
+if (has_any_capability( $capabilities, $systemcontext)) {
     $access = true;
 }
 
 if (!$access) {
-    // no access to coursecompletion report!
-    print_error('nopermissiontoviewcoursecompletionreport', 'error',  $CFG->wwwroot.'/my');
+    // No access to coursecompletion report!
+    print_error('nopermissiontoviewcoursecompletionreport', 'error', $CFG->wwwroot . '/my');
 }
 
 if (has_capability('report/coursecompletion:viewall', $systemcontext)) {
-    define('IS_ADMIN', true);
-    admin_externalpage_setup("reportcoursecompletion", "", null, "", array("pagelayout"=>"report"));
-
+    define('REPORT_COURSECOMPLETION_IS_ADMIN', true);
+    admin_externalpage_setup("reportcoursecompletion", "", null, "", array("pagelayout" => "report"));
 } else {
-    define("IS_ADMIN", false);
+    define("REPORT_COURSECOMPLETION_IS_ADMIN", false);
     $PAGE->set_pagelayout('report');
 }
 
-// The default column to sort by
-$default_sort = IS_ADMIN ? "u.firstname" : "c.fullname";
+// The default column to sort by.
+$defaultsort = REPORT_COURSECOMPLETION_IS_ADMIN ? "u.firstname" : "c.fullname";
 
-// Columns of the report
+// Columns of the report.
 $columns = array(
     "course",
     "timestarted",
     "timecompleted",
     "completionstatus",
 );
-// Add user columns if user is admin
-if (IS_ADMIN) array_unshift($columns, 'name', 'email');
+// Add user columns if user is admin.
+if (REPORT_COURSECOMPLETION_IS_ADMIN) {
+    array_unshift($columns, 'name', 'email');
+}
 
-// Sort sql fields for each column
+// Sort sql fields for each column.
 $scolumns = array(
     'course' => array('c.fullname'),
     'timestarted' => array('cc.timestarted'),
     'timecompleted' => array('cc.timecompleted'),
     'completionstatus' => array('completionstatus'),
 );
-// Add user columns if user is admin
-if (IS_ADMIN) {
-    $scolumns = array_merge(array(
+// Add user columns if user is admin.
+if (REPORT_COURSECOMPLETION_IS_ADMIN) {
+    $scolumns = array_merge(
+        array(
         'name' => array('u.firstname', 'u.lastname'),
         'email' => array('u.email')
-    ), $scolumns);
+        ),
+        $scolumns
+    );
 }
 
-// Variables that hold SQL
-$user_where   = "WHERE u.id = :userid";
-$user_params  = array('userid' => $USER->id);
+// Variables that hold SQL.
+$userwhere   = "WHERE u.id = :userid";
+$userparams  = array('userid' => $USER->id);
 $where        = "";
 $params       = [];
-$cohort_join  = "";
+$cohortjoin  = "";
 
-// Build array of all the possible sort columns
+// Build array of all the possible sort columns.
 $allsorts = array();
-foreach($scolumns as $sorts) {
-    foreach($sorts as $s) {
+foreach ($scolumns as $sorts) {
+    foreach ($sorts as $s) {
         $allsorts[] = $s;
     }
 }
 
-$sort = optional_param("sort", $default_sort, PARAM_NOTAGS); // Sorting column.
+$sort = optional_param("sort", $defaultsort, PARAM_NOTAGS); // Sorting column.
```

Comment:
looks like a potential SQL injection vulnerability here - sort is checked against allsorts above, but PARAM_NOTAGS is definitely the wrong param to be using here - PARAM_ALPHA or PARAM_TEXT would be safer to use.